crates/rattler/src/libsolv/solvable.rs

```diff
@@ -0,0 +1,113 @@
+use std::ffi::CStr;
+use std::ptr::NonNull;
+
+use crate::libsolv::ffi;
+use crate::libsolv::pool::{FindInterned, PoolRef, StringId};
+use crate::libsolv::repo::RepoRef;
+
+/// Solvable in libsolv
+#[repr(transparent)]
+pub struct Solvable(NonNull<ffi::Solvable>);
+
+/// Represents a solvable in a [`Repo`] or [`Pool`]
+#[derive(Copy, Clone)]
+pub struct SolvableId(pub(super) ffi::Id);
+
+impl From<SolvableId> for ffi::Id {
+    fn from(s: SolvableId) -> Self {
+        s.0
+    }
+}
+
+impl SolvableId {
+    /// Resolves to the interned type returns a Solvable
+    pub fn resolve(&self, pool: &PoolRef) -> Solvable {
+        // Safe because the new-type wraps the ffi::id and cant be created otherwise
+        unsafe {
+            // Re-implement pool_id2solvable, as it's a static inline function, we can't use it :(
+            let solvables = pool.as_ref().solvables;
+            // Apparently the solvable is offset by the id from the first solvable
+            let solvable = solvables.offset(self.0 as isize);
+            Solvable(NonNull::new(solvable).expect("solvable cannot be null"))
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct SolvableInfo {
```

Comment:
What is the difference between a `SolvableInfo` and `Solvable`? Couldn't you just add accessor methods for all of the fields inside this struct to `Solvable`?